### DISCUSSION TITLE
bgp advertisement and lb ipam

### DISCUSSION THREAD

**Discussion by wibed** [cilium/cilium#36247] [Q&A] [answered]
i dont get how bgp advertisement of type podcidr and lb ipam does exactly work together.

one has node restricted cidrs the other reaches across the cluster.
even if they are working side by side, how would they work together?
limitations, etc.

**Comment by harsimran-pabla** [comment, marked as answer]
- podcidr are advertised to enabled intra-cluster communication when Cilium is running in native routing mode. Upstream peering routers will receive pod cidrs from Cilium BGP, and will know how to route traffic between pods. 
- lb ipam is used for allocating IP to services of type load-balancer and Cilium BGP will advertise these VIPs to connected network. Traffic coming from outside the kubernetes cluster can reach service VIP and routed to one of the backend pods of the service.
